### PR TITLE
Compat: Test that sample_mask pipelines fail

### PR DESCRIPTION
**Pull request by greggman**
Note: you can run this test with this URL

[http://localhost:8080/standalone/?compatibility=1&runnow=1&q=compat:api,validation,render_pipeline,shader_module:sample_mask:*](http://localhost:8080/standalone/?compatibility=1&runnow=1&q=compat:api,validation,render_pipeline,shader_module:sample_mask:*)

You'll need a browser that supports compat like Chrome Canary with webgpu developer features enabled in flags

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/compat/api/validation/render_pipeline/shader_module.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        struct Output {
+            @builtin(sample_mask) mask_out: u32,
+            @location(0) color : vec4f,
+        }
+        @fragment fn fsWithoutSampleMaskUsage() -> @location(0) vec4f {
+            return vec4f(1.0, 1.0, 1.0, 1.0);
+        }
+        @fragment fn fsWithSampleMaskUsage() -> Output {
+            var o: Output;
+            // We need to make sure this sample_mask isn't optimized out even its value equals "no op".
```
Comment (lokokung) [inline src/compat/api/validation/render_pipeline/shader_module.spec.ts:40]: nit. even if its*

Comment (lokokung) [inline src/compat/api/validation/render_pipeline/shader_module.spec.ts:40]: nit. even if its*

File: src/compat/api/validation/render_pipeline/shader_module.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+export const description = `
+Tests that you can not create a render pipeline with a shader module that uses sample_mask in compat mode.
```
Comment (lokokung) [inline src/compat/api/validation/render_pipeline/shader_module.spec.ts:2]: nit. This is the same description as the only test. Maybe this can be a more general description for all tests (and potential tests) for the file as to not duplicate the information?

**Review by lokokung — APPROVED** [review]
Small nit, LGTM!

File: src/compat/api/validation/render_pipeline/shader_module.spec.ts (hunk 1 above)
Comment (greggman) [reply src/compat/api/validation/render_pipeline/shader_module.spec.ts:40]: done

Comment (greggman) [reply src/compat/api/validation/render_pipeline/shader_module.spec.ts:40]: done

File: src/compat/api/validation/render_pipeline/shader_module.spec.ts (hunk 2 above)
Comment (greggman) [reply src/compat/api/validation/render_pipeline/shader_module.spec.ts:2]: done
